rfcs/0136-stabilize-incrementally.md

```diff
@@ -0,0 +1,189 @@
+There is now a growing desire to stabilize those features, but as of yet no plan on exactly how to do so.
+The tension between that desire and a lack of a clear plan has loomed over our heads for a while; this RFC aims to dispel it by providing a concrete plan, an plan that hopefully will mitigate the lingering controversies and tensions around Flakes.
+
+## The situation so far
+
+There are a few facts that must be introduced to give context to this situation.
+We can't change these short of a time machine, so we must find a way to live wit them instead.x
+
+### Flakes are very popular
+
+As measured in the community polls, Nix has a lot of new users, dwarfing the number of long-time users.
+Flakes is very popular among these new users.
+Ergo, Flakes is very popular among the Nix community as a whole.
+
+Like it or not, these users have been using Flakes as if it was stable, and we cannot make huge drastic changes that would break their code in hard-to-fix ways.
+
+Many Groups and individuals interested in the continued growth the Nix community see Flakes are popular, an also wish it to be stabilized to attract further still more users, since Flakes are already proven to be popular among users.
+
+### Difficulties in the roll-out
+
+It is undeniable that the roll-out of these new features has been controversial.
+Graham Christensen's blog post [flakes-are-an-obviously-good-thing](https://grahamc.com/blog/flakes-are-an-obviously-good-thing) lays out the procedure issues along the way quite well.
+
+Some people were upset the Flakes RFC was abandoned but the feature merged.
+Other people were fine with experimental features being merged without RFC, but were upset because Flakes never really *felt* experimental.
+Experimental features would be modified a lot subject to community design use, and, most importantly, *discouraged* from being used in production.
+
+### Flakes are criticized for encroaching on other features
+
+There are many criticism about Flakes.
+But one of them especially relevant to stabilizing is a perception that Flakes have encroached on other new features, in the sense that it ought to be possible to use them without Flakes but isn't in practice.
+For example, there is no reason in theory pure eval requires Flakes, but without the ability to populate an initial whitelist of store paths that are safe to import it in practice does.
+
+This is especially noticeable for new CLI features that *previously did*, in fact, work without Flakes.
+For example, in earlier versions of Nix `nix search` worked without Flakes.
+
+## A plan all sides can be happy with
+
+Stabilizing the new CLI and Flakes will end the saga of the past few years.
+It is a last good chance to sooth some of these tensions and put us on a good foot moving forward.
+The new NixOS Foundation gives me hope Nix is serious about addressing these sorts of governance issues, and thus would be a good way to further demonstrate we are turning a new leaf.
```

Comment:
This is more of a style point, but I don't think an RFC should contain `I` sentences, i.e. sentences where the author is present in the first person.

In my opinion an RFC is different from a blog post in that regard, in that it's a place to propose an "object" for discussion, and this object is then evaluated by the community. It helps in this sense if the object is described in "objective" rather than "subjective" terms, removing unnecessary references to the author.

Basically more like a scientific paper than a discursive post.

Some might disagree and that's fine, just wanted to give a more meta-level feedback :)

Reply:
Yeah it's a hard problem where I don't want to "hog the floor" as the author but I don't want to appear as speak for others that might disagree.

Happy to do "I" -> "the authors" or something else, do propose suggestions!

Reply:
```suggestion
The new NixOS Foundation gives the RFC authors hope Nix is serious about addressing these sorts of governance issues, and thus would be a good way to further demonstrate we are turning a new leaf.
```